### PR TITLE
Add site_gen.sh script to help generate new sites

### PR DESCRIPTION
**Pull request by serngawy**
Signed-off-by: melserngawy <melserng@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,19 @@
 # ztp-acm-manifests
-Repository to host all manifests used in Zero Touch Deployment,
-to deploy an ACM cluster, enroll the spoke clusters, and create
+This repository to host all manifests used in Zero Touch Deployment (https://github.com/redhat-ztp/ztp-cluster-deploy) to deploy an ACM cluster, enroll the spoke clusters, and create
```
Comment (yrobla) [inline README.md]: The purpose of this repository is to host

File: README.md [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
 the needed subscriptions for configuring the clusters
+
+# For external use with Zero Touch Deployment
+- You will need to fork this repository and the RAN manifest repository at https://github.com/redhat-ztp/ztp-ran-manifests. Check the RAN manifest repository for more info about the RAN manifest.
```
Comment (yrobla) [inline README.md]: too repetitive... RAN manifest two times in the same sentence

File: README.md [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+
+- You will need to execute the [site_gen.sh](https://github.com/redhat-ztp/ztp-acm-manifests/blob/main/site_gen.sh) script to create your cluster profile directory.
+
+- The [site_gen.sh](https://github.com/redhat-ztp/ztp-acm-manifests/blob/main/site_gen.sh) script take 3 parameters with the following order to execute:
```
Comment (yrobla) [inline README.md]: take/takes

File: site_gen.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+CLUSTER_NAME=$1
+profile=$2
```
Comment (yrobla) [inline site_gen.sh]: why profile is lower case? PROFILE for consistency

File: site_gen.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+RAN_MANIFEST_REPO=$3
+ORG_RAN_MANIFEST_REPO=https://github.com/redhat-ztp/ztp-ran-manifests
+
+if [ $profile == "cu" ]; then
+  cp -rf subscriptions/sites/sample_site_cu_subscription subscriptions/sites/${CLUSTER_NAME}_subscription
+  sed -i "s/sample-site-cu/$CLUSTER_NAME/g" subscriptions/sites/${CLUSTER_NAME}_subscription/*
+elif [ $profile == "du" ]; then
+  cp -rf subscriptions/sites/sample_site_du_subscription subscriptions/sites/${CLUSTER_NAME}_subscription
```
Comment (yrobla) [inline site_gen.sh:12]: why not just replace the var? sample_site_${profile}  instead of the if... this will allow to add more profiles in the future

File: subscriptions/sites/sample_site_cu_subscription/02_sample_site_cu_subscription.yaml [hunk 6]
```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     app: sample-site-cu-subscription
   annotations:
-    apps.open-cluster-management.io/git-path: manifests/sites/sample_site_cu
+    apps.open-cluster-management.io/git-path: manifests/sites/sample-site-cu
```
Comment (yrobla) [inline subscriptions/sites/sample_site_cu_subscription/02_sample_site_cu_subscription.yaml:9]: why did you change that path?

**Review by yrobla — CHANGES_REQUESTED** [review]
comments/requests for change

File: site_gen.sh (hunk 5 above)
Comment (serngawy) [reply site_gen.sh:12]: just more restrict check for cu and du profile name and gives more flexibility for directory name

File: subscriptions/sites/sample_site_cu_subscription/02_sample_site_cu_subscription.yaml (hunk 6 above)
Comment (serngawy) [reply subscriptions/sites/sample_site_cu_subscription/02_sample_site_cu_subscription.yaml:9]: cause the site directory name in the ran manifest should match the cluster_name in the acm manifest. Otherwise it will replace all the sample_site_cu/du naming in kustomization.yaml

Comment (yrobla) [reply subscriptions/sites/sample_site_cu_subscription/02_sample_site_cu_subscription.yaml:9]: That's what i don't understand... in our case our path is https://github.com/redhat-ztp/ztp-ran-manifests/tree/main/manifests/sites/sample_site_cu . If you create a new site then, should not be https://github.com/redhat-ztp/ztp-ran-manifests/tree/main/manifests/sites/$CLUSTER_NAME ? so it points to manifests/sites/$CLUSTER_NAME path?

Comment (serngawy) [reply subscriptions/sites/sample_site_cu_subscription/02_sample_site_cu_subscription.yaml:9]: in the script the, the sed command is looking for the sample-site-cu and replace it with the new $CLUSTER_NAME . If the sed command also replace sample_site_cu it will also replace the file names in the kustomization.yaml which will required change the new cluster file names. I don't want to miss with that just keep it simple.

**Review by yrobla — APPROVED** [review]
